packages/botonic-plugin-flow-builder/src/action/first-interaction.ts

```diff
@@ -0,0 +1,78 @@
+import { INPUT } from '@botonic/core'
+
+import { BOT_ACTION_PAYLOAD_PREFIX, MAIN_FLOW_NAME } from '../constants'
+import { FlowContent } from '../content-fields'
+import { getNodeByUserInput } from '../user-input'
+import { FlowBuilderContext } from './index'
+import { getContentsByKnowledgeBase } from './knowledge-bases'
+import { getContentsByPayload } from './payload'
+
+export async function getContentsByFirstInteraction({
+  cmsApi,
+  flowBuilderPlugin,
+  request,
+  resolvedLocale,
+}: FlowBuilderContext): Promise<FlowContent[]> {
+  const firstInteractionContents =
+    await flowBuilderPlugin.getStartContents(resolvedLocale)
+
+  if (request.input.data && request.input.type === INPUT.TEXT) {
```

Comment:
i'm wondering if it could make sense to have a util function for this like `textInputHasData` or similar, as it is a check used in many places

Reply:
inputWithTextData?